packages/backend/src/managers/environmentManager.ts

```diff
@@ -0,0 +1,134 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { type PodInfo, type Webview, containerEngine } from '@podman-desktop/api';
+import type { PodmanConnection } from './podmanConnection';
+import { LABEL_RECIPE_ID } from './applicationManager';
+import { MSG_ENVIRONMENTS_STATE_UPDATE } from '@shared/Messages';
+import type { EnvironmentState } from '@shared/src/models/IEnvironmentState';
+
+export class EnvironmentManager {
+  #environments: Map<string, EnvironmentState>;
+
+  constructor(
+    private webview: Webview,
+    private podmanConnection: PodmanConnection,
+  ) {
+    this.#environments = new Map();
+  }
+
+  adoptRunningEnvironments() {
+    this.podmanConnection.startupSubscribe(() => {
+      if (!containerEngine.listPods) {
+        // TODO(feloy) this check can be safely removed when podman desktop 1.8 is released
```

Comment:
can we open an issue to keep track of this and the other similar TODO?

Reply:
right, created #242